### ISSUE TITLE
`odo dev` fails component was deployed using `odo deploy`

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#5546]
testing with https://github.com/kadel/devfile-nodejs-deploy

```
$ odo deploy
...
...

$ odo dev
Using devfile.yaml from the current directory.
Starting your application on cluster in developer mode
 ✗  Waiting for Kubernetes resources [4s]
 ✗  failed to update config to component deployed.: multiple Deployments exist for the selector: app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app. Only one must be present

```

**Comment by cdrage** [comment]
I believe I've fixed this in: https://github.com/redhat-developer/odo/pull/5468

**Comment by feloy** [comment]
Since we are now setting the labels on both resources deployed on DEV and DEPLOY mode, I think we need to include the mode label to check for unicity, here:
- app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app,odo.dev/mode=Deploy
- app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app,odo.dev/mode=Dev

**Comment by kadel** [comment]
> I believe I've fixed this in: #5468

it is not, while testing that PR i found this.
But this was not introduced there, it is in main branch as well, this is why i created separate issue.

**Comment by kadel** [comment]
> Since we are now setting the labels on both resources deployed on DEV and DEPLOY mode, I think we need to include the mode label to check for unicity, here:
> 
> * app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app,odo.dev/mode=Deploy
> * app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app,odo.dev/mode=Dev

exactly, when `odo dev` is checking for resources it needs to include `odo.dev/mode=dev` label, and when `odo deploy` checking for resources it needs to include `odo.dev/mode=deploy`

**Comment by cdrage** [comment]
> Since we are now setting the labels on both resources deployed on DEV and DEPLOY mode, I think we need to include the mode label to check for unicity, here:
> 
> * app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app,odo.dev/mode=Deploy
> * app.kubernetes.io/instance=devfile-nodejs-deploy,app.kubernetes.io/part-of=app,odo.dev/mode=Dev

This has been done in the `odo list` refactor for https://github.com/redhat-developer/odo/pull/5468 but I'm unsure why it's still failing. I'll look into this.
